Docs11112/storage monitoring restructure (#29706)
DataDog/documentation: content/en/integrations/guide/storage-monitoring-setup.md

This guide explains how to configure Storage Monitoring in Datadog for your Amazon S3 buckets, Google Cloud Storage buckets, and Azure storage accounts. Access your Storage Monitoring data by navigating to **Infrastructure > Storage Monitoring**.

Select your cloud storage service to access setup instructions.

{{< partial name="cloud_storage_monitoring/storage-monitoring-setup.html" >}}

## Setup for Amazon S3

{{< tabs >}}
{{% tab "Recommended: Storage Monitoring UI" %}}

The fastest way to set up Storage Monitoring is going to **Infrastructure > Storage Monitoring > [Add Buckets][1]**. On the Add Buckets page, you can configure multiple S3 buckets for Storage Monitoring in one go.

1. Go to Datadog > Infrastructure > Storage Monitoring.
2. Click [Add Buckets][1].

{{< img src="integrations/guide/storage_monitoring/add-buckets.png" alt="Select buckets for enabling Storage Monitoring" responsive="true">}}

3. Enable Amazon S3 Integration and Resource collection for all the AWS accounts you want to monitor.

>    **Note**: For each AWS account containing S3 buckets to monitor, ensure your Datadog IAM role includes the `s3:GetObject`, `s3:ListObjects`, and `s3:PutInventoryConfiguration` permissions.

Reviewer: Could we make this part a little more clear? I'd like to emphasize that datadog only needs those permissions for the inventory bucket, not all of their buckets. Maybe something like:

**Note:** Ensure Datadog is able to access your inventory files by adding the `s3:GetObject` and `s3:ListBucket` permissions to the Datadog IAM integration role for the accounts which own those buckets. These read-only permissions should be scoped to only the inventory destination buckets

Example:

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "AllowListInventoryBucket",
      "Effect": "Allow",
      "Action": "s3:ListBucket",
      "Resource": "arn:aws:s3:::storage-monitoring-s3-inventory-destination"
    },
    {
      "Sid": "AllowGetInventoryObjects",
      "Effect": "Allow",
      "Action": "s3:GetObject",
      "Resource": "arn:aws:s3:::storage-monitoring-s3-inventory-destination/*"
    }
  ]
}
```

Additionally, ensure that the destination buckets' policies allow your source bucket to write inventory configurations

See https://docs.aws.amazon.com/AmazonS3/latest/userguide/configure-inventory.html#configure-inventory-destination-bucket-policy for more info

Storage Monitoring for Amazon S3, Google Cloud Storage, and Azure Blob Storage provides deep, prefix-level analytics to help you understand exactly how your storage is being used. It detects potential issues before they impact operations, and helps you make data-driven decisions about storage optimization. Use these insights to track storage growth, investigate access patterns, and optimize costs.

This guide explains how to configure Storage Monitoring in Datadog for your Amazon S3 buckets, Google Cloud Storage buckets, and Azure storage accounts. Access your Storage Monitoring data by navigating to **Infrastructure > Storage Monitoring**.

Select your cloud storage service to access setup instructions.

{{< partial name="cloud_storage_monitoring/storage-monitoring-setup.html" >}}

## Setup for Amazon S3

{{< tabs >}}
{{% tab "Recommended: Storage Monitoring UI" %}}

> The fastest way to set up Storage Monitoring is going to **Infrastructure > Storage Monitoring > [Add Buckets][1]**. On the Add Buckets page, you can configure multiple S3 buckets for Storage Monitoring in one go.

Reviewer 2: I know it's not part of your changes, but wanted to clean this up a bit while I was at it. 

```suggestion
The fastest way to configure Storage Monitoring is through the **Add Buckets** page in Datadog, where you can set up multiple S3 buckets at the same time.
```

1. Go to Datadog > Infrastructure > Storage Monitoring.
2. Click [Add Buckets][501].

{{< img src="integrations/guide/storage_monitoring/add-buckets.png" alt="Select buckets for enabling Storage Monitoring" responsive="true">}}

3. Enable Amazon S3 Integration and Resource collection for all the AWS accounts you want to monitor.

>    **Note**: Ensure Datadog is able to access your inventory files by adding the `s3:GetObject` and `s3:ListBucket` permissions to the Datadog IAM integration role for the accounts which own those buckets. These read-only permissions should be scoped to only the inventory destination buckets. 
>    
>    The destination buckets' policies also must allow your source bucket to write inventory configurations. See [Creating a destination bucket policy][502] in the AWS documentation for more information.
> 
>    Example source-bucket policy:

Reviewer 2: This is a bit dense. Maybe we can break out into sub-steps for clarity?

```suggestion
   1. **Allow Datadog to read from your destination buckets.** Add the following permissions to the Datadog IAM integration role for the account that owns the destination buckets:
      - `s3:GetObject`
      - `s3:ListBucket`
      
      Scope these read-only permissions to only the destination buckets containing your S3 inventory files.
      
   1. **Allow source buckets to write to destination buckets.** The destination buckets must include a policy that allows the source buckets to write inventory data. See [Creating a destination bucket policy][502] in the AWS documentation for details.
   
   Example source-bucket policy:
```

[105]: https://forms.gle/L97Ndxr2XLen1GBs7
[106]: https://aws.amazon.com/s3/pricing/
{{% /tab %}}

{{% tab "Terraform" %}}

You can use the Terraform `aws_s3_bucket_inventory` resource to set up Storage Monitoring. 

> The example below shows how to enable daily inventory on an S3 bucket for Datadog monitoring. To use this example:

Reviewer 2: ```suggestion
The following example shows how to enable daily inventory on an S3 bucket for Datadog monitoring. To use this example:
```

   - Replace `<MY_MONITORED_BUCKET>` with the name of the bucket to be monitored.
   - Replace `<MY_INVENTORY_DESTINATION>` with the name of the bucket that receives your inventory files.
   - Replace `<DESTINATION_ACCOUNT_ID>` with the AWS account ID that owns the destination bucket.

```tf
resource "aws_s3_bucket" "monitored" {
  bucket = "<MY_MONITORED_BUCKET>"
}

resource "aws_s3_bucket" "inventory_destination" {
  bucket = "<MY_INVENTORY_DESTINATION>"
}

resource "aws_s3_bucket_inventory" "daily_inventory" {
  bucket = aws_s3_bucket.monitored.id
  name   = "datadog-daily-inventory"

  included_object_versions = "All"
  schedule {
    frequency = "Daily"
  }
  destination {
    bucket {
      account_id = "<DESTINATION_ACCOUNT_ID>"
      bucket_arn = aws_s3_bucket.inventory_destination.arn
      format     = "CSV"
      prefix     = "datadog-inventory/"
    }
  }
  optional_fields = [
    "Size",
    "StorageClass",
    "LastModifiedDate"
  ]
}
```

**Notes**:

>    - The destination bucket can be your source bucket, but for security/logical separation, many organizations use a separate bucket.

Reviewer 2: ```suggestion
   - The destination bucket can be your source bucket, but for security and logical separation, many organizations use a separate bucket.
```

[401]: https://docs.google.com/forms/d/e/1FAIpQLScd0xor8RQ76N6BemvvMzg9UU7Q90svFrNGY8n83sMF2JXhkA/viewform
[402]: https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/tree/master/examples/s3-inventory
{{% /tab %}}

{{% tab "AWS Console" %}}

To manually set up the required [Amazon S3 Inventory][206] and related configuration, follow these steps:

### Step 1: Create a destination bucket

1. [Create an S3 bucket][201] to store your inventory files. This bucket acts as the central location for inventory reports. **Note**: You must only use one destination bucket for all inventory files generated in an AWS account.
2. Create a prefix within the destination bucket (optional).

### Step 2: Configure the bucket and integration role policies

1. Follow the steps in the [Amazon S3 user guide][202] to add a bucket policy to your destination bucket allowing write access (`s3:PutObject`) from your source buckets.

2. Ensure the Datadog AWS integration role has `s3:GetObject` and `s3:ListObjects` permissions on the destination bucket. These permissions allow Datadog to read the generated inventory files.

> ### Step 3: Configure Inventory generation

Reviewer 2: Not sure if "inventory" is considered a proper noun here but it's lowercase in in step 3 ("Click Create inventory configuration.") so following that here.

```suggestion
### Step 3: Configure inventory generation
```

- Use only one destination bucket for all inventory files per Google Cloud project
- Verify all permissions are correctly configured
- If issues persist, [reach out][1] with your bucket details, Google Cloud Project ID, and Datadog org ID

[1]: mailto:storage-monitoring@datadoghq.com
[2]: https://cloud.google.com/storage/docs/insights/using-inventory-reports#enable_the_api
[3]: https://forms.gle/c7b8JiLENDaUEqGk8

## Setup for Azure Blob Storage

{{< tabs >}}
{{% tab "Azure CLI" %}}

Enable inventories for the selected storage accounts in each subscription by running the following script in your [Azure Cloud Shell][301]:

```shell
curl https://datadogstoragemonitoring.blob.core.windows.net/scripts/install.sh \
  | bash -s -- <CLIENT_ID> <SUBSCRIPTION_ID> <COMMA_SEPARATED_STORAGE_ACCOUNT_NAMES>
```

Before running the script, set your [shell environment][302] to Bash and replace the various placeholder inputs with the correct values:
- `<CLIENT_ID>`: The client ID of an App Registration already set up using the [Datadog Azure integration][302]
- `<SUBSCRIPTION_ID>`: The subscription ID of the Azure subscription containing the storage accounts

> - `<COMMA_SEPARATED_STORAGE_ACCOUNT_NAMES>`: A comma-separated list of the storage accounts you want to monitor. For example, `storageaccount1,storageaccount2`

Reviewer 2: So we don't have a rogue period in between these two fragments that's inconsistent with no periods for bullets 1 and 2.

```suggestion
- `<COMMA_SEPARATED_STORAGE_ACCOUNT_NAMES>`: A comma-separated list of the storage accounts you want to monitor (for example, `storageaccount1,storageaccount2`)
```
